### DISCUSSION TITLE
How immediate is synchronization supposed to be?

### DISCUSSION THREAD

**Discussion by wsw70** [dani-garcia/vaultwarden#1926] [Q&A] [answered]
I use VW behind a Caddy v2 reverse proxy with the configuration from the documentation (for a year+ - great software, thanks!)

It seems to me that synchronization (between devices or between a device and the server) is not instantaneous. It happens eventually, though.

When is a synchronization triggered? Is it when I save an entry (new or existing)? I realize that this is a question about *clients* and not the VW server, but maybe someone will know :)

How can I debug the synchronization? I did not see any logs in VW, should I rather rely on the logs of the proxy?

**Comment by BlackDex** [comment]
See https://github.com/dani-garcia/vaultwarden/wiki/Enabling-WebSocket-notifications

**Reply by wsw70** [reply]
Thank you @BlackDex 

I just checked "browser → chrome extension" and "chrome extension → browser"  synchronization - it is instantenous.

From the wiki you linked to:

> Note that WebSocket notifications are not applicable to the mobile (Android/iOS) Bitwarden clients. These clients use the native push notification service instead (FCM for Android, APNs for iOS). vaultwarden does not currently support push notifications to mobile clients.

Does that mean that an item will be synchronized from/to the mobile client only manually (by forcing the synchronization)?

I see that my tests were partly synchronized (the first ones did synchronize, the last one not (yet)). I could not find any setting on the client for a cyclic synch bit maybe this is a built-in feature.

**Comment by jjlin** [comment, marked as answer]
In the absence of WebSocket or mobile push notifications, the clients sync periodically, or when certain events occur.

For the browser extensions, a sync runs each time you log in or unlock. If the extension stays unlocked, it syncs in the background every 6 hours:

https://github.com/bitwarden/browser/blob/294e7c3b8d8e7ae6d71433ea4305ce37f8ec3c3f/src/background/main.background.ts#L693

For the mobile clients, a sync runs each time the app is started or resumed, unless one has already run within the last 30 minutes:

https://github.com/bitwarden/mobile/blob/6ca5b66aa7accee64815111066bc801eb273cf7c/src/App/App.xaml.cs#L393
